src/js/structs/Batch.js

```diff
@@ -55,6 +54,28 @@ class Batch {
    * @param {Object} item
    */
   add(item) {
+    let batchIndex = this.length;
+
+    // As the while is faster then the Array.prototype.forEach and this
+    // function can potentially be called more often we choose while here.
+    while (--batchIndex >= 0) {
+      let {path, value} = this[batchIndex] || {};
+      let hasEqualPaths = item && Array.isArray(path) &&
+        Array.isArray(item.path) && path.join() === item.path.join();
+      if (hasEqualPaths) {
+        // Last entry with same path holds same value,
+        // do not add redundant action
+        if (value === item.value) {
+          return;
+        }
+        // Remove previous if path is the same as current to minimize
+        // number of actions
+        if (batchIndex === this.length - 1) {
+          this.pop();
+        }
+      }
+    }
```

Comment:
We talked offline and decided to remove the look-back to remove older values as there might be added stuff in between that makes this action invalid, e.g. with duplicable fields, which might change one field path into another.